etcdserver/server.go

```diff
@@ -323,9 +326,8 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		return nil, fmt.Errorf("unsupported bootstrap config")
 	}
 
-	err = os.MkdirAll(cfg.MemberDir(), privateDirMode)
-	if err != nil && err != os.ErrExist {
-		return nil, err
+	if mkerr := fileutil.MkdirAllAlways(cfg.MemberDir()); mkerr != nil {
```

Comment:
In theory, it should not. 

If the member dir does not exist before, it must be writable after we create it.

If the member dir does exist before, we checked at line 294.

But it is no harm to check again and make the code more readable, I will add.